AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1898,6 +1904,7 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
   if ((_methodOverrides & ASDisplayNodeMethodOverrideLayoutSpecThatFits) || _layoutSpecBlock != NULL) {
     ASLayoutSpec *layoutSpec = [self layoutSpecThatFits:constrainedSize];
     layoutSpec.parent = self; // This causes upward propogation of any non-default layoutable values.
+    layoutSpec.traitCollection = self.asyncTraitCollection;
```

Comment:
@rcancro the idea I had in mind with an object and struct based representation is that the common path internal implementation would always use the struct, and the object would exist almost soley for API surface area that third party code accesses.  This would mean that very few nodes create the object - important because there are a truly vast number of nodes, due to the intentional design decision where ASDataController relies on them being cheap, and creates entire cell subtrees for every datasource element.

If that makes sense, we should probably change this usage point / layoutSpec to use environment.  This is the only clear case I see that would "accidentally" trigger the creation, as the rest are in test project code (perfect & legitimate usages because they are "third party" code simulations).